lib/elixir/test/elixir/calendar/datetime_test.exs

```diff
@@ -941,6 +941,25 @@ defmodule DateTimeTest do
                  microsecond: {123_456, 6}
                }
     end
+
+    test "add/2 with non erlang time_units" do
```

Comment:
Added this test to ensure the code ends up with the actual intended result down the code path